core/controllers/contributor_dashboard.py

```diff
@@ -260,6 +263,45 @@ def _content_in_review(self, state_name, content_id, suggestions):
             s.change.content_id == content_id for s in suggestions)
 
 
+class MachineTranslatedTextHandler(base.BaseHandler):
+    """Provide a machine translation of exploration content."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests."""
+        exp_id = self.request.get('exp_id')
+        state_name = self.request.get('state_name')
+        content_ids = json.loads(self.request.get('content_ids'))
+        target_language_code = self.request.get('target_language_code')
+
+        # TODO(#12341): Tidy up this logic once we have a canonical list of
+        # language codes.
+        if not utils.is_supported_audio_language_code(
+                target_language_code
+            ) and not utils.is_valid_language_code(
+                target_language_code
+            ):
+            raise self.InvalidInputException(
+                'Invalid target_language_code: %s' % target_language_code)
+
+        translated_texts = {}
+        for content_id in content_ids:
+            translated_text = (
+                translation_services.get_machine_translation_for_content_id(
+                    exp_id, state_name, content_id, target_language_code
+                )
+            )
+            if translated_text is not None:
+                translated_texts[content_id] = translated_text
+        if translated_texts:
+            self.values = {'translated_texts': translated_texts}
+            self.render_json(self.values)
+        else:
+            raise self.PageNotFoundException()
```

Comment:
Should it be an empty dict? It seems to me that the cases where 1 vs 0 translations out of 20 were successfully provided should be handled the same way.

Reply:
**Not** Fixed. I plan on doing some refactoring here for efficiency and debugging.

Reply:
Refactor complete. I moved the get_machine_translation_for_content_id logic into MachineTranslatedStateTextHandler and updated the return values and tests accordingly.